Duckling/Numeral/DE/Rules.hs

```diff
@@ -323,15 +215,23 @@ ruleIntegerWithThousandsSeparator = Rule
       _ -> Nothing
   }
 
+ruleAllNumeralWords :: Rule
+ruleAllNumeralWords = Rule
+  { name = "simple and complex numerals written as one word"
+  , pattern = [regex "(ein|zwei|drei|vier|fünf|sech|sieb|acht|neun|zehn|elf|zwölf|hundert|tausend)?([^\\s]+)?(eine[m|n|r|s]?|eins?|zwei|drei|vier|fünf|sechs|sieben|acht|neun|zehn|elf|zwölf|[s|ß|z]ig|hundert|tausend)"]
+  , prod = \tokens -> case tokens of
+      (Token RegexMatch (GroupMatch matches):_) ->
+        (parseNumeral $ concat $ Text.unpack . Text.toLower <$> matches)
+        >>= integer
```

Comment:
```suggestion
  , prod = \ (Token RegexMatch (GroupMatch matches) : _)
    -> (parseNumeral $ concat $ Text.unpack . Text.toLower <$> matches)
        >>= integer
```

No need for the lambda + case here, we can just destructure the argument

Reply:
It's my understanding that ```prod``` is only applied if the regex pattern of the rule is met. But then the lambda restriction```\ (Token RegexMatch (GroupMatch matches) : _)``` will also always be met. So my understanding is that pattern match is total. Therefore, I suggest to adopt @stroxler's suggested change.

Reply:
GHC will give a warning, so you have not convinced GHC it is total. There is no compile time proof that the rules provided will only produce a RegexMatch. Not providing a wildcard is just a recipe for a later developer to get a confusing pattern match exception at runtime. I would strongly advise to just add the one line.

Reply:
prod does not run only if the pattern match supplied in prod itself occurs. The caller of prod decides what is passed to it - pattern matches do not restrict their domain in this way.

Reply:
I see. I'll add the elsewhere case. Many thanks, @chessai and @stroxler, for your kind support. This helps me a lot.